workflows/bcellmagic.nf

```diff
@@ -182,64 +188,107 @@ workflow BCELLMAGIC {
     ch_software_versions = ch_software_versions.mix(FASTQC.out.version.first().ifEmpty(null))
 
     // Merge UMI from index file to R1 if provided
-    if (params.index_file) {
-        MERGE_UMI ( ch_merge_umi_gunzip )
-        .set{ ch_gunzip }
-    } else {
-        RENAME_FASTQ ( ch_merge_umi_gunzip )
-        .set{ ch_gunzip }
-    }
+    if (params.umi) {
```

Comment:
Let me know if you still prefer to split this out into separate workflows. I think it is looking reasonably tidy with the subworkflows and [single `if (umi_length == 0)` check](https://github.com/dladd/bcellmagic/blob/e298b60b66d90bdef7d58d877d3bb136ab106eca/workflows/bcellmagic.nf#L169) and avoids maintaining the non-presto bits in duplicate. I'm also not sure if the separate workflow strategy would/should have been instead of or in addition to the separate subworkflow route.